activerecord/test/cases/scoping/relation_scoping_test.rb

```diff
@@ -228,6 +228,14 @@ def test_current_scope_does_not_pollute_sibling_subclasses
       assert SpecialComment.all.any?
     end
   end
+
+  def test_circular_joins_with_current_scope_does_not_crash
+    assert_nothing_raised do
```

Comment:
Like this? https://github.com/rails/rails/pull/25702/commits/5e6d8241f9a979fa248f9736fc64834ac86d4b5c